master/internal/experiment/bulk_action.go

```diff
@@ -734,3 +735,90 @@ func MoveExperiments(ctx context.Context,
 	}
 	return results, nil
 }
+
+func changeExperimentConfigLogRetention(ctx context.Context, database db.DB,
+	expID int, numDays int16,
+) error {
+	exp, err := db.ExperimentByID(ctx, expID)
+	if err != nil {
+		return errors.Wrap(err, "fetching experiment from database")
+	}
+
+	if !model.TerminalStates[exp.State] {
+		return status.Error(codes.FailedPrecondition, fmt.Sprintf(
+			"experiment in non terminal state '%s', try again later", exp.State))
+	}
+
+	activeConfig, err := database.ActiveExperimentConfig(exp.ID)
+	if err != nil {
+		return errors.Wrapf(
+			err, "unable to load config for experiment %v", exp.ID,
+		)
+	}
+	activeConfig.SetLogRetentionDays(&numDays)
+
+	if err := database.SaveExperimentConfig(exp.ID, activeConfig); err != nil {
+		return errors.Wrapf(err, "patching experiment %d", exp.ID)
+	}
+
+	return nil
+}
+
+// RetainLogs retains logs for the given list of experiments.
+func RetainLogs(ctx context.Context, database db.DB,
+	expIDs []int32, filters *apiv1.BulkExperimentFilters, numDays int16,
+) ([]ExperimentActionResult, error) {
+	var results []ExperimentActionResult
+	editableExperimentIDList, err := editableExperimentIds(ctx, expIDs, filters)
+	if err != nil {
+		return nil, err
+	}
+	if filters == nil {
+		for _, originalID := range expIDs {
+			if !slices.Contains(editableExperimentIDList, originalID) {
+				results = append(results, ExperimentActionResult{
+					Error: api.NotFoundErrs("experiment", fmt.Sprint(originalID), true),
+					ID:    originalID,
+				})
+			}
+		}
+	}
+
+	var intExpIDs []int
+	for _, v := range editableExperimentIDList {
+		err = changeExperimentConfigLogRetention(ctx, database, int(v), numDays)
+		if err != nil {
+			results = append(results, ExperimentActionResult{
+				Error: err,
+				ID:    v,
+			})
+		} else {
+			intExpIDs = append(intExpIDs, int(v))
+			results = append(results, ExperimentActionResult{
+				Error: nil,
+				ID:    v,
+			})
+		}
+	}
+
+	_, taskIDs, err := db.ExperimentsTrialAndTaskIDs(ctx, db.Bun(), intExpIDs)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to gather trial IDs for experiments")
+	}
+
+	if len(taskIDs) > 0 {
```

Comment:
style nite: indent the exception flow, so handle
```
if len(taskIDs) == 0 {
    return results, nil
}

... the rest of the flow that is now intended.
```
this also makes code easier to refactor later.

https://google.github.io/styleguide/go/decisions#indent-error-flow

Reply:
Done!